src/main/modules/ardoq/YarnParser.ts

```diff
@@ -1,28 +1,34 @@
-import { HTTPError } from '../../HttpError';
-
 import { Dependency } from './Dependency';
 import { IParser } from './IParser';
 
 import yaml from 'js-yaml';
 
+const { Logger } = require('@hmcts/nodejs-logging');
+
 export class YarnParser implements IParser {
   public extractTopTierDeps(depString: string): Dependency[] {
     const doc = yaml.load(depString) as Record<string, YamlDependency>;
 
+    const logger = Logger.getLogger('YarnParser');
+
     try {
       return Object.entries(doc)
-        .filter(([key]) => key !== '__metadata')
-        .map(([key, value]) => {
+        .filter(([key, value]) => {
+          if (key === '__metadata') {
+            return false;
+          }
           if (value.version === undefined) {
-            throw new HTTPError(`Failed to parse yarn.lock file. Failed parsing key: ${key} value: ${value}`, 400);
+            logger.warn(`Failed to parse yarn.lock file. Failed parsing key: ${key} value: ${value}`);
+            return false;
           }
-          return new Dependency(key, value.version);
-        });
+          return true;
+        })
+        .map(([key, value]) => new Dependency(key, value.version));
     } catch (e) {
       if (e.message === 'Cannot convert undefined or null to object') {
         return [];
       }
-      throw new HTTPError(`Failed to parse yarn.lock file. ${e.message}`, 400);
+      throw e;
```

Comment:
I think you can remove this whole try catch statement if you change line 20 to:
```
if (value?.value === undefined)
```

You can updated your `YamlDependency` interface to be more correct and say that `version` is sometimes undefined and then the compiler will help you.

Reply:
just re-read this, sorry. The try catch is actually there for when `Object.entries(doc)` throws up the `Cannot convert undefined or null to object` error due to say a blank yaml file. It's a bit annoying the library doesn't give specific error type I can handle for that, it's just a TypeError when I go to try and use it :(